overlord/restart/restart_context.go

```diff
@@ -0,0 +1,97 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package restart
+
+import (
+	"github.com/snapcore/snapd/boot"
+	"github.com/snapcore/snapd/bootloader"
+	"github.com/snapcore/snapd/logger"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/release"
+	"github.com/snapcore/snapd/snap"
+)
+
+type RestartContext struct {
```

Comment:
I'm not entirely sure if context is a good name for this, maybe PendingSystemRestart?

Reply:
I agree actually, I have stripped 'performReboot' from RestartContext as it doesn't actually need to be a member of RestartContext, so now `RestartContext` is primarily just a bearer of parameters for rebooting (and a implementation of the RebootBootloader interface). So I was thinking maybe we should call it `RestartOptions` or `RestartParameters` now, or something in that direction? Otherwise I'd be happy to call it PendingSystemRestart

Reply:
RestartParameters seems fine

Reply:
awesome, renamed now